test/e2e/field_selectors/field_selectors_test.go

```diff
@@ -0,0 +1,225 @@
+package field_selectors
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/flightctl/flightctl/test/harness/e2e"
+	"github.com/flightctl/flightctl/test/login"
+	"github.com/flightctl/flightctl/test/util"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	deviceYAMLPath  = "examples/device.yaml"
+	fleetYAMLPath   = "examples/fleet.yaml"
+	repoYAMLPath    = "examples/repository-flightctl.yaml"
+	unknownSelector = "unknown or unsupported selector"
+	resourceCreated = `(200 OK|201 Created)`
+)
+
+// Define the struct for parsing device.yaml
+type Device struct {
+	Metadata struct {
+		Name string `yaml:"name"`
+	} `yaml:"metadata"`
+}
+
+func TestFieldSelectors(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Field Selectors E2E Suite")
+}
+
+var _ = Describe("Field Selectors in Flight Control", func() {
+	var (
+		harness *e2e.Harness
+	)
+
+	BeforeEach(func() {
+		harness = e2e.NewTestHarness()
+		login.LoginToAPIWithToken(harness)
+	})
+
+	AfterEach(func() {
+		harness.Cleanup(false)
+	})
+
+	// Helper function to dynamically extract device name
+	extractDeviceName := func() string {
+		deviceName, err := extractNameFromYAML(filepath.Join(util.GetTopLevelDir(), deviceYAMLPath))
+		Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to extract name from %s", deviceYAMLPath))
+		return strings.TrimSpace(deviceName)
+	}
+
+	Context("login", func() {
+		It("should have worked, and we can list devices", func() {
+			out, err := harness.CLI("get", "devices")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(ContainSubstring("NAME"))
+		})
+	})
+
+	Context("apply/fleet", func() {
+		It("create a complete fleet", func() {
+			_, _ = harness.CLI("delete", "fleet")
+			out, err := harness.CLI("apply", "-f", filepath.Join(util.GetTopLevelDir(), fleetYAMLPath))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(MatchRegexp(resourceCreated))
+
+		})
+	})
+
+	Context("apply/recursive", func() {
+		It("create a device", func() {
+			_, _ = harness.CLI("delete", "device")
+			deviceName := extractDeviceName()
+			out, err := harness.CLI("apply", "-R", "-f", filepath.Join(util.GetTopLevelDir(), deviceYAMLPath))
+			time.Sleep(30 * time.Second) //to Establish fleet before adding device to it
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(MatchRegexp(resourceCreated))
+			Expect(out).To(ContainSubstring(fmt.Sprintf("%s/%s", deviceYAMLPath, deviceName)))
+		})
+	})
+
+	Context("apply/repo", func() {
+		It("create a complete repo", func() {
+			out, err := harness.CLI("apply", "-f", filepath.Join(util.GetTopLevelDir(), repoYAMLPath))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(MatchRegexp(resourceCreated))
+		})
+
+	})
+
+	Context("Basic Functionality Tests", Label("77917"), func() {
```

Comment:
Could you please Use "It" for test case and "By" for the test steps? Otherwise each step will be reported as a test